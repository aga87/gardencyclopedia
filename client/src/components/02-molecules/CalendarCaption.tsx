import React from 'react';

type CalendarCaptionProps = {
  noOfPlants: number;
};

const CalendarCaption = ({ noOfPlants }: CalendarCaptionProps): JSX.Element => (
  <div className='m-calendar-caption'>
    <details className='m-calendar-caption__details'>
      <summary className='m-calendar-caption__details-summary'>
        {noOfPlants} {noOfPlants === 1 ? 'Plant' : 'Plants'}
      </summary>
      <table>
        <caption className='m-calendar-caption__legend-caption'>Legend</caption>
        <thead>
          <tr>
            <th className='m-calendar-caption__legend-heading'>Sow</th>
            <th className='m-calendar-caption__legend-heading'>Harvest</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className='m-calendar-caption__legend-sow' />
            <td className='m-calendar-caption__legend-harvest' />
          </tr>
        </tbody>
      </table>
    </details>
  </div>
);

export default CalendarCaption;
