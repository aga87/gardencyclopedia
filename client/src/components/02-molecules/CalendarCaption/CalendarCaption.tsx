import React from 'react';

type CalendarCaptionProps = {
  noOfPlants: number;
};

const CalendarCaption = ({ noOfPlants }: CalendarCaptionProps): JSX.Element => (
  <div className='m-calendar-caption'>
    <details className='m-calendar-caption__details'>
      <summary className='m-calendar-caption__details-summary'>
        {noOfPlants} {noOfPlants === 1 ? 'Plant' : 'Plants'}
      </summary>
      <table>
        <caption className='m-calendar-caption__legend-caption'>Legend</caption>
        <thead>
          <tr>
            <th className='m-calendar-caption__legend-heading'>Sow</th>
            <th className='m-calendar-caption__legend-heading'>Harvest</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className='m-calendar-caption__legend-sow'
              aria-label='Green color'
            />
            <td
              className='m-calendar-caption__legend-harvest'
              aria-label='Yellow color'
            />
          </tr>
        </tbody>
      </table>
    </details>
  </div>
);

export default CalendarCaption;
