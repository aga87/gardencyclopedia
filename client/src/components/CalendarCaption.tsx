import React from 'react';

type CalendarCaptionProps = {
  noOfPlants: number;
};

const CalendarCaption = ({ noOfPlants }: CalendarCaptionProps): JSX.Element => (
  <figcaption className='c-calendar-caption'>
    <details className='c-calendar-caption__details'>
      <summary className='c-calendar-caption__details-summary'>
        {noOfPlants} {noOfPlants > 1 ? 'Plants' : 'Plant'}
      </summary>
      <table>
        <tr>
          <th className='c-calendar-caption__legend-heading'>Sow</th>
          <th className='c-calendar-caption__legend-heading'>Harvest</th>
        </tr>
        <tr>
          <td className='c-calendar-caption__legend-sow' />
          <td className='c-calendar-caption__legend-harvest' />
        </tr>
      </table>
    </details>
  </figcaption>
);
export default CalendarCaption;
