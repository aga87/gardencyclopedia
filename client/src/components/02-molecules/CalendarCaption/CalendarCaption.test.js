import { render, cleanup, getByRole } from '@testing-library/react';
import CalendarCaption from './CalendarCaption';

afterEach(cleanup);

const defaultProps = {
  noOfPlants: 12
};

test('CalendarCaption renders with correct text', () => {
  const { getByText } = render(<CalendarCaption noOfPlants={0} />);
  expect(getByText('0 Plants')).toBeInTheDocument();
  cleanup();
  render(<CalendarCaption noOfPlants={1} />);
  expect(getByText('1 Plant')).toBeInTheDocument();
  cleanup();
  render(<CalendarCaption {...defaultProps} />);
  expect(getByText('12 Plants')).toBeInTheDocument();
});

test('CalendarCaption renders the legend with no visual regression', () => {
  const { getByText, container } = render(
    <CalendarCaption {...defaultProps} />
  );
  expect(getByText('Legend')).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});
