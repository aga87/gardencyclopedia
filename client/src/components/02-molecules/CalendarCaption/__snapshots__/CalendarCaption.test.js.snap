// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarCaption renders the legend with no visual regression 1`] = `
<div
  class="m-calendar-caption"
>
  <details
    class="m-calendar-caption__details"
  >
    <summary
      class="m-calendar-caption__details-summary"
    >
      12
       
      Plants
    </summary>
    <table>
      <caption
        class="m-calendar-caption__legend-caption"
      >
        Legend
      </caption>
      <thead>
        <tr>
          <th
            class="m-calendar-caption__legend-heading"
          >
            Sow
          </th>
          <th
            class="m-calendar-caption__legend-heading"
          >
            Harvest
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="m-calendar-caption__legend-sow"
          />
          <td
            class="m-calendar-caption__legend-harvest"
          />
        </tr>
      </tbody>
    </table>
  </details>
</div>
`;
